Wordle.Api/Wordle.Api/Controllers/WordController.cs

```diff
@@ -24,4 +28,67 @@ public async Task<string> GetWordOfDay(double offsetInHours = -7.0)
         DateOnly today = DateOnly.FromDateTime(DateTime.UtcNow.AddHours(offsetInHours));
         return await wordOfTheDayService.GetWordOfTheDay(today);
     }
-}
+
+    [Authorize(Roles = Roles.Admin)]
+    [HttpGet("WordOfTheDayHint")]
+    public async Task<string> GetWordOfDayHint(double offsetInHours = -7.0)
+    {
+        DateOnly today = DateOnly.FromDateTime(DateTime.UtcNow.AddHours(offsetInHours));
+        var wordOfTheDay = await wordOfTheDayService.GetWordOfTheDay(today);
+
+        return wordOfTheDay.Substring(0, 1) + "___" + wordOfTheDay.Substring(4, 1);
+    }
+
+    [HttpGet("SecuredRandomWord")]
+    [Authorize(Policy = Policies.RandomAdmin)]
+    public async Task<string> GetSecuredRandomWord()
+    {
+        var randomWord = await wordOfTheDayService.GetRandomWord("sec");
+        return randomWord.Text;
+    }
+
+
+    [HttpGet("WordByDate/{month}/{day}/{year}")]
+    public async Task<string?> GetWordByDate(int month, int day, int year)
+    {
+        DateOnly date = new DateOnly(year, month, day);
+        return await wordOfTheDayService.GetWordOfTheDay(date);
+    }
```

Comment:
```suggestion
    [HttpGet("WordByDate/{date}")]
    public async Task<string?> GetWordByDate(Datetime date)
    {
        DateOnly date = DateOnly.FromDateTime(date);
        return await wordOfTheDayService.GetWordOfTheDay(date);
    }
```
Why are using `int` values to representing date and then converting it to a date when you could a query parameter that is of type `DateTime`?

Reply:
It seemed simpler to me than using DateTime. Wouldn't you have to set all of the extra fields that you don't need? Or is there a Javascript object/method that does that for you?

Reply:
You can utilize the regular Date() object with formatting. There's ways to convert the Date to the correct format of mm-dd-yyyy or any other formats if necessary.